src/search.cpp

```diff
@@ -187,9 +190,35 @@ Searcher::Searcher(Searcher&& other) = default;
 Searcher& Searcher::operator=(Searcher&& other) = default;
 Searcher::~Searcher() = default;
 
+namespace
+{
+
+bool archivesAreEquivalent(const Archive& a1, const Archive& a2)
+{
+  // XXX: Can we use the UUID to check the equivalence of two ZIM archives?
+  // NOTE: If any of the two archives was initialized from a file-descriptor
+  // NOTE: then they are considered non-equivalent.
```

Comment:
We can use the uuid yes.
By definition uuid is here to identify a ZIM archive, so we should use it
And it would be better that the path/filename which is a way to locate the archive.

Reply:
Done